apriltag.c

```diff
@@ -1001,6 +1006,12 @@ zarray_t *apriltag_detector_detect(apriltag_detector_t *td, image_u8_t *im_orig)
     if (td->wp == NULL || td->nthreads != workerpool_get_nthreads(td->wp)) {
         workerpool_destroy(td->wp);
         td->wp = workerpool_create(td->nthreads);
+        if(td->wp == NULL){
+            // creating workerpool failed
+            printf("apriltag.c: Creating workerpool failed.");
```

Comment:
Is it possible to not print text on stdout in the core library? This should be controlled by a logger that can be turned on/off. Maybe we can also use `perror` as in the case below.

What are the reasons for failing to create a worker pool? Can we forward this error to the user via the `errno`?

Reply:
The call to `pthread_create` in `workerpool_create` can fail if the system cannot provide all the threads requested (50,000 for instance) - in doing so it sets the `errno` to `EAGAIN`

So checking for `EAGAIN` after `apriltag_detector_detect()` should show that something has gone wrong.

Reply:
If `errno` is set already, can we skip the `printf`? I see that there are many `printf` in other places, but I think those should also be replaced in the long-term with something the user can control.

I suggest that we use `errno` to indicate errors and `perror` if required to print on `stderr`. I would avoid printing to `stdout`.